src/sql-parser/src/parser.rs

```diff
@@ -3716,6 +3717,21 @@ impl<'a> Parser<'a> {
         }
     }
 
+    fn parse_kafka_include_header(&mut self) -> Result<SourceIncludeMetadata, ParserError> {
+        let key = self.parse_literal_string()?;
+        let alias = self
+            .parse_keyword(AS)
+            .then(|| self.parse_identifier())
+            .transpose()?;
+        let use_bytes = self.parse_keyword(BYTES);
+        let header = KafkaHeader { key, use_bytes };
```

Comment:
I find it a bit confusing that the structure of parsing this is like `[HEADER PART 1] [ALIAS] [HEADER PART 2]` instead of `[HEADER] [ALIAS]` or `[ALIAS] [HEADER]`. It also complicates the `AstDisplay` implementation. Is there a specific reason for this structure?

Reply:
I'm confused about what you're confused about, Joe!

Reply:
`SourceIncludeMetadata` has two fields, a `SourceIncludeMetadataType` and an `Option<Ident>`. In this function, to parse a `SourceIncludeMetadata` we take the following steps in order:
1. Parse a `String` (`key`) for the `SourceIncludeMetadataType`.
2. Parse the `Option<Ident>` (`alias`).
3. Parse a keyword (`use_bytes`) for the `SourceIncludeMetadataType`.

That means that we cannot display a `SourceIncludeMetadataType` without knowing the context that it is used in. If you look at the `AstDisplay` implementation of `SourceIncludeMetadataType`, we don't display the full contents of `SourceIncludeMetadataType::Header(_)`. In the `AstDisplay` implementation of `SourceIncludeMetadata`, we display the rest of the contents of the inner `SourceIncludeMetadataType::Header`.

If there's a specific reason for that structure, then I think it's fine. Otherwise, it's generally simpler if `SourceIncludeMetadataType` is responsible for displaying it's entire contents.

Reply:
That part of the code is indeed a bit awkward. And you are right, because the`HEADER` syntax is `INCLUDE HEADER <name> AS <ident> [BYTES]` it parses the `name`, `ident`, and `[BYTES]` into `<SourceIncludeMetadataType> <SourceIncludeMetadata> <SourceIncludeMetadataType::Header>`. And therefore `SourceIncludeMetadataType` needs to display things before and after `SourceIncludeMetadata` it cannot display it's entire contents with the current structure of the code.

I think the cleanest solution would indeed be to adapt the syntax to something like `INCLUDE HEADER <name> [FORMAT [BYTES | TEXT]] AS <ident>`. And semantically it makes more sense as well. 

Given that the Webhook is relatively new (and it looks like we need to adapt the syntax anyhow), it might be easier to adapt the Webhook syntaxt to `INCLUDE HEADER [FORMAT [BYTES | TEXT]] AS header_name` or if we are concerned with backward compatibility to `INCLUDE HEADER [FORMAT [BYTES | TEXT]] AS header_name | AS header_name BYTES`.

Reply:
The AST should follow the SQL syntax, not the other way around! This suggests to me that we need to adjust the `SourceIncludeMetadata` AST node to instead look like:

```rust
#[derive(Debug, Clone, PartialEq, Eq, Hash)]
pub enum SourceIncludeMetadata {
    Key { alias: Option<Ident> },
    Timestamp { alias: Option<Ident> },
    Partition { alias: Option<Ident> },
    Offset { alias: Option<Ident> },
    Headers { alias: Option<Ident>, bytes: bool },
}

impl AstDisplay for SourceIncludeMetadata {
    fn fmt<W: fmt::Write>(&self, f: &mut AstFormatter<W>) {
        let print_alias = |alias| match alias {
            None => (),
            Some(alias) => {
                f.write_str(" AS ");
                f.write_node(alias);
            }
        };
        match self {
            SourceIncludeMetadata::Key { alias } => {
                f.write_str("KEY");
                print_alias(alias);
            }
            SourceIncludeMetadata::Timestamp => {
                f.write_str("TIMESTAMP");
                print_alias(alias);
            }
            SourceIncludeMetadata::Partition => {
                f.write_str("PARTITION");
                print_alias(alias);
            }
            SourceIncludeMetadata::Offset => {
                f.write_str("OFFSET");
                print_alias(alias);
            }
            SourceIncludeMetadata::Headers => {
                f.write_str("HEADERS");
                print_alias(alias);
                if bytes {
                    f.write_str(" BYTES")
                }
            }
        }
    }
}
```

> I think it might be a bit too late to change the webhook syntax since it's out of private preview. Though, I'll delegate to @ParkMyCar any questions about changing the webhook syntax.

I personally like the current webhook syntax. I could see the argument for allowing `FORMAT BYTES` in addition to `BYTES`, but I think we should defer that to another PR so we can discuss the merits of that change independently of this change.

Reply:
>The AST should follow the SQL syntax, not the other way around!

This is on me as I had suggested it. Even though the `alias` now has to be matched within each metadata type separately I guess this would be easier to read and understand what belongs where implicitly.